pkg/manager/tidbdashboard/manager.go

```diff
@@ -414,22 +428,24 @@ func generateTiDBDashboardService(td *v1alpha1.TidbDashboard) *corev1.Service {
 }
 
 func dashboardStartArgs(
+	listenHost string,
 	port int,
 	featureVersion, pathPrefix string,
-	clusterTLSEnable, mysqlTLSEnable, telemetry, experimental bool,
+	clusterTLSEnable, mysqlTLSEnable, telemetry, experimental, keyVisualizer bool,
 	tc *v1alpha1.TidbCluster,
 ) []string {
 	pdAddress := fmt.Sprintf("%s.%s:%d", controller.PDMemberName(tc.Name), tc.Namespace, v1alpha1.DefaultPDClientPort)
 
 	base := []string{
-		"-h=0.0.0.0",
+		fmt.Sprintf("-h=%s", listenHost),
 		fmt.Sprintf("-p=%d", port),
 		fmt.Sprintf("--data-dir=%s", dataPVCMountPath),
 		fmt.Sprintf("--temp-dir=%s", dataPVCMountPath),
 		fmt.Sprintf("--path-prefix=%s", pathPrefix),
 		fmt.Sprintf("--feature-version=%s", featureVersion),
 		fmt.Sprintf("--experimental=%t", experimental),
 		fmt.Sprintf("--telemetry=%t", telemetry),
+		fmt.Sprintf("--keyVisualizer=%t", keyVisualizer),
```

Comment:
how about only appending this arg if it needs to be disabled, then we can avoid recreating the existing Pod when upgrading the TiDB Operator.

Reply:
@csuzhangxc, done. Please re-review